### PR TITLE
Add aarch64 build support

### PR DESCRIPTION
**Pull request by martin-g**
Checklist
* [x] Used a [personal fork of the feedstock to propose changes](https://conda-forge.org/docs/maintainer/updating_pkgs.html#forking-and-pull-requests)
* [ ] Bumped the build number (if the version is unchanged)
* [ ] Reset the build number to `0` (if the version changed)
* [ ] [Re-rendered]( https://conda-forge.org/docs/maintainer/updating_pkgs.html#rerendering-feedstocks ) with the latest `conda-smithy` (Use the phrase <code>@<space/>conda-forge-admin, please rerender</code> in a comment in this PR for automated rerendering)
* [ ] Ensured the license file is being packaged.

### CODE REVIEW

**Comment by martin-g** [comment]
This PR is a clone of https://github.com/conda-forge/blis-feedstock/pull/27 + one commit (https://github.com/conda-forge/blis-feedstock/pull/30/commits/c9b665928662831019ab775bf50e1a818bf38d82) as requested by @isuruf at https://github.com/conda-forge/blis-feedstock/pull/29#issuecomment-2039819264

**Comment by h-vetinari** [comment]
@conda-forge-admin, please rerender

**Review by h-vetinari — APPROVED** [review]
This basically LGTM, but I'll leave the final approval to @isuruf

File: recipe/build.sh [hunk 1]
```diff
@@ -4,34 +4,65 @@ CFLAGS=$(echo "${CFLAGS}" | sed "s/-mtune=[a-zA-Z0-9]*//g")
...
+
+# General case
+./configure --prefix=$PREFIX --enable-cblas --enable-threading="$MODEL" $EXTRA $arch
+make -j${CPU_COUNT}
+make install
+make check -j${CPU_COUNT}
+
+
+# Windows-specific shenanigans (builds twice; first static above, then shared below)
+case $target_platform in win-*)
```
Comment (martin-g) [inline recipe/build.sh:61]: Is this possible ?
I thought that `bld.bat` would be used for Windows, not `build.sh`.

**Comment by martin-g** [comment]
The Windows build fails with:
```
2024-04-10T23:09:41.6845616Z kernels/sandybridge/3/bli_gemm_sandybridge_int_d8x4.c:96:2: error: use of undeclared identifier '__m256d'
2024-04-10T23:09:41.6846019Z    96 |         __m256d va0_3b_0, va4_7b_0;
2024-04-10T23:09:41.6846240Z       |         ^
2024-04-10T23:09:41.6846740Z kernels/sandybridge/3/bli_gemm_sandybridge_int_d8x4.c:97:2: error: use of undeclared identifier '__m256d'
2024-04-10T23:09:41.6847111Z    97 |         __m256d va0_3b_1, va4_7b_1;
2024-04-10T23:09:41.6847415Z       |         ^
```

**Comment by martin-g** [comment]
Let me compare it against https://github.com/conda-forge/blis-feedstock/pull/29 !
Since the sources are the same then it must be some config file (i.e. the re-rendering is to blame indeed).

**Comment by martin-g** [comment]
It must be 
https://github.com/conda-forge/blis-feedstock/pull/30/files#diff-283d4846cd6e9d12c8444a622943204d7a168d0c8c350127106345fd3f42ac2bR5-R6

these two lines are missing at https://github.com/conda-forge/blis-feedstock/pull/29/files#diff-283d4846cd6e9d12c8444a622943204d7a168d0c8c350127106345fd3f42ac2b

Should I remove them manually ? Or you want to take a deeper look at the re-rendering ?

**Comment by h-vetinari** [comment]
> Should I remove them manually ? Or you want to take a deeper look at the re-rendering ?

Manually removing this is not sustainable, because the next rerender of the recipe will re-add it. To make it permanent, we need https://github.com/conda-forge/conda-forge-pinning-feedstock/pull/5756, where I'm waiting for feedback from @isuruf.

That said, I'm very surprised this has any effect at all; in particular, the vs-related pieces in the build environment are the same both in the failing CI run (_with_ the `c_stdlib_version` on windows) as well as the passing one:
```
    ucrt:                  10.0.22621.0-h57928b3_0        conda-forge
    vc:                    14.3-hcf57466_18               conda-forge
    vc14_runtime:          14.38.33130-h82b7239_18        conda-forge
    vs2015_runtime:        14.38.33130-hcb4865c_18        conda-forge
    vs2019_win-64:         19.29.30139-he1865b1_18        conda-forge
    vs_win-64:             2019.11-he1865b1_18            conda-forge
    vswhere:               3.1.4-h57928b3_0               conda-forge
```

**Comment by martin-g** [comment]
https://github.com/conda-forge/conda-forge-pinning-feedstock/pull/5756 has been merged!
Please let me know if you want me to do something here (revert the last commit, re-render, ...) !

**Comment by martin-g** [comment]
@isuruf Friendly ping!

**Pull request merged**